File: intellij/src/main/kotlin/ai/jolli/jollimemory/core/WindowsHidden.kt
package ai.jolli.jollimemory.core

import java.nio.file.Files
import java.nio.file.Path

/** Best-effort `dos:hidden` on Windows; no-op elsewhere. Silent on failure — hidden bit is cosmetic. */
object WindowsHidden {
    private val isWindows: Boolean =
        System.getProperty("os.name").lowercase().contains("win")

    fun tryMarkHidden(path: Path) {
        if (!isWindows) return
        try {
            Files.setAttribute(path, "dos:hidden", true)
        } catch (_: Exception) {
            /* hidden bit is cosmetic — never fail the caller */
        }
    }
}

File: cli/src/util/WindowsHidden.ts
import { platform } from "node:os";
import { spawnSyncHidden } from "./Subprocess.js";

/** Best-effort `attrib +h` on win32; no-op elsewhere. Silent on failure — hidden bit is cosmetic. */
export function tryMarkHiddenOnWindows(absPath: string): void {
	if (platform() !== "win32") return;
	try {
		spawnSyncHidden("attrib", ["+h", absPath], {
			timeout: 2000,
		});
	} catch {
		/* hidden bit is cosmetic — never fail the caller */
	}
}
